sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/UserIdentity.java

```diff
@@ -193,7 +193,7 @@ private static void validateProviderLogin(String providerLogin) {
 
     private static void validateLogin(String login) {
       checkArgument(isNotBlank(login), "User login must not be blank");
-      checkArgument(login.length() <= 255 && login.length() >= 3, "User login size is incorrect (Between 3 and 255 characters)");
+      checkArgument(login.length() <= 255 && login.length() >= 2, "User login size is incorrect (Between 2 and 255 characters)");
```

Comment:
I think it should be 2 constants that are called from non API code. So even if we duplicate the check logic we have the same limits